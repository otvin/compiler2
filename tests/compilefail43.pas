program compilefail43(output);
{illegal use of global variable as the control variable of a for statement}

var i:integer;

procedure foo(j:integer);
var k:integer;
begin

	for k := 0 to j do begin
		writeln(k);
	end;

	for i := 15 downto j do begin
		writeln(i);
	end;
end;

begin

	foo(5);
	foo(2);
	foo(-7);
	foo(3);

end.
