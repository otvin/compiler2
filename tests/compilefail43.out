Control variable 'i' must be declared in same block as the 'for' in file:tests/compilefail43.pas line:14    column:9  
